app/policies/proposal_policy.rb

```diff
@@ -0,0 +1,32 @@
+class ProposalPolicy
+  include TreePolicy
+  attr_reader :user, :proposal
+
+  def perm_trees
+    {
+      edit?: [:author?, :not_approved?],
+      update?: :edit?
+    }
+  end
+
+  def initialize(user, proposal)
+    @user = user
+    @proposal = proposal
+  end
+
+  def author?
+    @proposal.requester_id == @user.id
```

Comment:
Minor: if you're using `attr_reader`, access the attributes with `self.ATTRNAME`. If you spell instance variables wrong, they give back `nil`, which is really annoying, so I tend to use them as little as possible.